### ISSUE TITLE
CentOS 8.2 install experience/issues

### ISSUE THREAD

**Issue by captainhook** [apisnetworks/apnscp#35]
A few issues I have experienced when trying to install on CentOS 8.2 on Azure.

The machine comes with /etc/resolv.conf containing an Azure IP for DNS. Bootstrap process appears to replace this with 127.0.0.53. My install command uses 'Cloudflare' for DNS without an API key at this point.

I can add the Azure nameserver (or any other valid resolver) back into resolv.conf and resume the installation.

Once install is resumed, I see:
```
TASK [system/kernel : Enumerate kernels] ****************************************************************************
fatal: [localhost]: FAILED! => changed=false
  cmd: |-
    awk -F\' '/^menuentry / {print $2}' /boot/grub2/grub.cfg | cat -n | awk '{print $1-1,$1="",$0}' | grep -F "4.18.0-193.6.3.el8_2" | awk  '{print $1}'
  delta: '0:00:00.014156'
  end: '2020-06-26 18:43:49.758071'
  failed_when_result: true
  rc: 0
  start: '2020-06-26 18:43:49.743915'
  stderr: ''
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

```

After first reboot:

`2020-06-26 18:48:36,323 p=1276 u=root n=dnf.[fork.13666] | Unknown configuration value: failovermethod=priority in /etc/yum.repos.d/pgdg-redhat-all.repo; Configuration: OptionBinding with id "failovermethod" does not exist`

So I have to again add the valid Azure nameserver back into resolv.conf and it appears to run fine then. This happened again 1 hour into install.

Running `systemd-resolve --status` shows:
```
Global
       LLMNR setting: yes
MulticastDNS setting: yes
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
  Current DNS Server: 1.0.0.1
         DNS Servers: 1.0.0.1
                      1.1.1.1
          DNS Domain: ~.
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
```

It may work better to utilise resolvconf to set DNS.

After installing/logging in, testing Argos relay displays error:
`Util_Process::formatDataCallProc(): ntfy: ERROR: Failed to open /root/.argos.conf Traceback (most recent call last): File "/usr/local/lib/python3.6/site-packages/ntfy/config.py", line 32, in load_config config = safe_load(open(expanduser(config_path))) FileNotFoundError: [Errno 2] No such file or directory: '/root/.argos.conf' `

**Comment by msaladna** [comment]
Thank you for the bug report. Fixed in [ab67aa07](https://gitlab.com/apisnetworks/apnscp/-/commit/ab67aa072beffe84074a2eb59c3835091753cfc5#49e15433807cad75db783e1f1c36a764f3aef601).

systemd introduces systemd-resolved in CentOS 8.1+, a caching stub resolver like unbound. This setting is preserved on install if configured. Bootstrapper applies the smallest set of changes to make a platform compliant with ApisCP; if systemd-resolved's stub resolver is active, then it's used. Ideally the image maintainers should disable this service if it's not in use.

"resolveconf" is an Ubuntu feature and not the appropriate way to manage DNS resolvers on RedHat systems. Resolver changes are covered in [common/update-config](https://gitlab.com/apisnetworks/apnscp/-/blob/master/resources/playbooks/roles/common/update-config/tasks/main.yml#L3-51) including systemd-resolved detection.

> TASK [system/kernel : Enumerate kernels]

CentOS 8 uses [BLS](https://www.freedesktop.org/wiki/Specifications/BootLoaderSpec/) to organize kernels. ApisCP doesn't know how to handle these just yet. The only downside is that it cannot automatically update kernels and reboot a server. This error is completely benign.

> Configuration: OptionBinding with id "failovermethod" does not exist

This is due to a bad directive in the stock [PostgreSQL repo](https://yum.postgresql.org/repopackages.php). I've opened an issue with their repo maintainers. The error however is non-fatal.

From what I've [tested](https://github.com/apisnetworks/apiscp-bootstrapper#provider-stats), Azure performance is terrible. Unless you have credits to burn I'd deploy on a more performant infrastructure.

**Comment by captainhook** [comment]
Thank you, that was superfast!

> From what I've tested, Azure performance is terrible.

On an Azure VM with 2 cores and 4GB memory, this install took well over 2 hours, similar to your test. I'm surprised it's so bad. I'll probably shift to OVH/Hetzner/Vultr once I've depleted these credits :)

Happy for this to be closed unless you want me to test anything.
Thanks.
